src/OMSimulatorLib/ComponentFMUCS.cpp

```diff
@@ -379,6 +384,12 @@ oms_status_enu_t oms::ComponentFMUCS::initializeDependencyGraph_initialUnknowns(
     {
       for (size_t j = startIndex[i]; j < startIndex[i + 1]; j++)
       {
+        if (dependency[j] <= 0 || allVariables.size() < dependency[j])
+        {
+          logError("Out of range of vector allVariables.");
```

Comment:
Should this rather be a debug message?

Reply:
If I return an `oms_status_fatal` this should be an error log I guess.

Reply:
But then you have to change the error message. The error messages are for the users, not for us :) And there is already the other error message that will be shown:
`Couldn't initialize dependency graph for initial unknowns.`